app/controllers/application_controller.rb

```diff
@@ -1,10 +1,150 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
+   register Sinatra::ActiveRecordExtension
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+  get '/' do
+
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in?
+      redirect("/tweets")
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if !params["username"].empty? && !params["email"].empty? && !params["password"].empty?
+      @user = User.create(params)
+      session[:user_id] = @user.id
+      redirect("/tweets")
+    else
+     redirect("/signup")
+   end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = current_user
+      erb :'/users/show'
+    else
+      redirect("/login")
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      @user = current_user
+      erb :'/tweets/create_tweet'
+    else
+      redirect("/login")
+    end
+  end
+
+  post '/tweets' do
+    if logged_in? && !params["content"].empty?
+      @tweet = Tweet.create(content: params["content"])
+      @tweet.user_id = current_user.id
+      @tweet.save
+      redirect("/tweets")
+    else
+      redirect("/tweets/new")
+    end
+  end
+
+  get '/tweets/:id' do
+    @tweet = Tweet.find_by_id(params["id"])
+    if logged_in?
+      erb :'/tweets/show_tweet'
+    else
+      redirect("/login")
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    @tweet = Tweet.find_by_id(params["id"])
+    if logged_in?
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?